src/software/backend/simulation/simulator_robot_test.cpp

```diff
@@ -3,39 +3,688 @@
+        createWorld(robot_non_zero_state, ball_zero_state);
+    EXPECT_EQ(simulator_robot->getDribblerTemperatureDegC(), 25);
+}
+
+TEST_F(SimulatorRobotTest, test_kick)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Point dribbling_point = getDribblingPoint(robot.position(), robot.orientation());
+    Ball ball(dribbling_point, Vector(0, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    // Simulate for 1/2 second without kicking
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Make sure we didn't kick
+    EXPECT_LT(simulator_ball->velocity().length(), 0.001);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+
+    simulator_robot->kick(5.0);
+
+    // Simulate for 1/2 second after kicking
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(5, 0)).length(), 0.005);
+}
+
+TEST_F(SimulatorRobotTest, test_chip)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Point dribbling_point = getDribblingPoint(robot.position(), robot.orientation());
+    Ball ball(dribbling_point, Vector(0, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    // Simulate for 1/2 second without chipping
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    // Make sure we didn't chip
+    EXPECT_LT(simulator_ball->velocity().length(), 0.001);
+    EXPECT_LT((simulator_ball->position() - dribbling_point).length(), 0.01);
+
+    simulator_robot->chip(5.0);
+
+    // Simulate for 1/2 second after chipping
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
+    }
+
+    EXPECT_GT(simulator_ball->velocity().x(), 1);
+    EXPECT_NEAR(simulator_ball->velocity().y(), 0, 0.001);
+}
+
+TEST_F(SimulatorRobotTest, test_robot_does_not_kick_ball_when_autokick_disabled)
+{
+    Robot robot(0, Point(0, 0), Vector(0, 0), Angle::zero(), AngularVelocity::zero(),
+                Timestamp::fromSeconds(0));
+    Ball ball(Point(0.15, 0), Vector(-0.25, 0), Timestamp::fromSeconds(0));
+    auto [world, simulator_robot, simulator_ball] = createWorld(robot, ball);
+
+    EXPECT_LT((simulator_ball->velocity() - Vector(-0.25, 0)).length(), 0.001);
+
+    // Simulate for 1/2 second
+    for (unsigned int i = 0; i < 30; i++)
+    {
+        world->stepSimulation(Duration::fromSeconds(1.0 / 60.0));
     }
+
+    EXPECT_LT(simulator_ball->velocity().length(), 0.25);
+}
+
```

Comment:
Should add tests for autokick disable autochip and vice-versa here once that's changed above.

Reply:
done

Reply:
Sorry, where did this get added?

Reply:
pushed, now should show up

Reply:
Sorry, I probably just can't see it (massive test lol), but where are we checking this?

Reply:
There are tests `test_enabling_autokick_disables_autochip` and vice-versa

Reply:
I'm super confused, I did see that but we don't seem to be checking if autochip is disabled in that test?

Reply:
Something like `isAutokickEnabled()` is not part of the public class interface, so we check the motion of the ball to determine if a kick or chip occured.

Reply:
I can change that

Reply:
done